packages/runtime/runtime-utils/src/serializer.ts

```diff
@@ -26,16 +71,49 @@ export class FluidSerializer implements IFluidSerializer {
     }
 
     public get IFluidSerializer() { return this; }
+    public get IFluidHandleEncoder() { return this; }
 
-    public replaceHandles(
-        input: any,
-        bind: IFluidHandle,
-    ) {
+    /**
+     * Given a mostly-jsonable object that may have handle objects embedded within, will return a fully-jsonable object
+     * where any embedded IFluidHandles have been replaced with a serializable form.
+     *
+     * The original `input` object is not mutated.  This method will shallowly clones all objects in the path from
+     * the root to any replaced handles.  (If no handles are found, returns the original object.)
+     */
+    public encode(input: any, bind: IFluidHandle) {
```

Comment:
Since these methods are considerable more efficient, should we update the `stringify` and `parse` to use these as well?

Reply:
I'm resurrecting my benchmarks to see how things have changed between node 10 and node 14.

Using JSON.parse(JSON.stringify(...)) and vice-versa to encode/decode handles is still unforgiveable (~10x-200x slower depending on complexity)

encode + stringify vs stringify w/a replacer is now narrow (~1x-5x, with 1x being typical.)

parse + decode vs. decode w/a replacer is now ~2x-5x (I predict 2x will be typical.)